lapis/src/main/antlr/org/genspectrum/lapis/model/advancedqueryparser/AdvancedQuery.g4

```diff
@@ -0,0 +1,132 @@
+grammar AdvancedQuery;
+
+// parser rules
+
+start: expr EOF;
+expr:
+  both             # Uni
+  | not_ expr         # Not
+  | expr and_ expr    # And
+  | expr or_ expr    # Or
+  | '(' expr ')'     # Parenthesis
+  | maybeQuery       # Maybe
+  ;
+
+both: 
+  single
+  | metadataQueryExpr
+  ;
+
+single:
+  nucleotideMutationQuery
+  | nOfQuery
+  | nucleotideInsertionQuery
+  | namedMutationQuery
+  | namedInsertionQuery
+  ;
+
+metadataQueryExpr:
+  metadataQuery
+  | metadataGreaterThanEqualQuery
+  | metadataLessThanEqualQuery
+  | isNullQuery
+  ;
+
+variantExpr:
+  single # VariantUni
+  | not_ variantExpr # VariantNot
+  | variantExpr and_ variantExpr # VariantAnd
+  | variantExpr or_ variantExpr # VariantOr
+  | '(' variantExpr ')'  # VariantParenthesis
+  | maybe_ '(' variantExpr ')' # VariantMaybe
+  ;
+
+maybeQuery:
+  maybe_ '(' variantExpr ')'
+  ;
+
+or_: OR | '|';
+maybe_: M A Y B E;
+not_: NOT | '!';
+and_: AND | '&';
+position: NUMBER;
+
+nucleotideMutationQuery : nucleotideMutationQueryFirstSymbol? position nucleotideMutationQuerySecondSymbol?;
```

Comment:
Nitpick: `singleSegmentedMutationQuery` would be more precise, because the current name doesn't really fit in the multi segmented case.